pkg/cluster/k8sres.go

```diff
@@ -1000,20 +1010,31 @@ func (c *Cluster) getPodEnvironmentSecretVariables() ([]v1.EnvVar, error) {
 	}
 
 	for k := range secret.Data {
-		secretPodEnvVarsList = append(secretPodEnvVarsList,
-			v1.EnvVar{Name: k, ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: c.OpConfig.PodEnvironmentSecret,
+		if !isEnvVarExist(secretPodEnvVarsList, k) {
+			secretPodEnvVarsList = append(secretPodEnvVarsList,
+				v1.EnvVar{Name: k, ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: c.OpConfig.PodEnvironmentSecret,
+						},
+						Key: k,
 					},
-					Key: k,
-				},
-			}})
+				}})
+		}
 	}
 
 	return secretPodEnvVarsList, nil
 }
 
+func isEnvVarExist(envs []v1.EnvVar, key string) bool {
```

Comment:
Minor naming thing: `is` would read better with an adjective, for instance `isEnvVarPresent`

Reply:
Good point, thanks, will change